block/block.go

```diff
@@ -113,6 +113,9 @@ func (m *Manager) applyBlock(block *types.Block, commit *types.Commit, blockMeta
 			return fmt.Errorf("commit block: %w", err)
 		}
 
+		// update last block time
+		m.State.SetLastBlockTime(block.Header.GetTimestamp())
```

Comment:
I think there is a case here that we crash after `ExecuteBlock` but before this statement. 
in that case the next block will have `isBlockAlreadyApplied` true, and we won't set the last block time for this block.

I recommmend moving this function to be at the end of the function just beore we return nil or at least after the if/else clause.

Reply:
moved